data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java

```diff
@@ -59,43 +58,57 @@ public SimpleDataService(
         this.dataReader = dataReader;
     }
 
-    public CompletableFuture<Optional<DataReaderRequest>> authoriseRequest(final DataRequest dataRequest) {
+    /**
+     * Query for the references.  It will return the information needed to retrieve the resources.  If there is no
+     * data to be returned, a {@link ForbiddenException} is thrown.
+     *
+     * @param dataRequest data provided by the client for requesting the resource
+     * @return reference to the resources that are to be returned to client
+     */
+    public CompletableFuture<AuthorisedData> authoriseRequest(final DataRequest dataRequest) {
         LOGGER.debug("Querying persistence for token {} and resource {}", dataRequest.getToken(), dataRequest.getLeafResourceId());
-        CompletableFuture<Optional<AuthorisedRequestEntity>> futureRequestEntity = this.persistenceLayer.getAsync(dataRequest.getToken(), dataRequest.getLeafResourceId());
+        CompletableFuture<Optional<AuthorisedRequestEntity>> futureRequestEntity = persistenceLayer.getAsync(dataRequest.getToken(), dataRequest.getLeafResourceId());
         return futureRequestEntity.thenApply(maybeEntity -> maybeEntity.map(
-                entity -> new DataReaderRequest()
-                        .context(entity.getContext())
-                        .user(entity.getUser())
-                        .resource(entity.getLeafResource())
-                        .rules(entity.getRules())
-                )
+                entity -> AuthorisedData.Builder.create()
+                        .withResource(entity.getLeafResource())
+                        .withUser(entity.getUser())
+                        .withContext(entity.getContext())
+                        .withRules(entity.getRules())
```

Comment:
I did go this way, but backed off when I realised that the AuthorisedRequestEntity is from a separate repo.  I thought it would be better to  build it by the parts so I would need to have an external dependency in the model class.  You look closer to the constion of the DataReadeRequest and you can see same problem but in reverse.  I would have to change DataReadeRequest in the reader repo to make this work along the same way.